core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -67,9 +70,11 @@ public void start(final EventListener eventListener, AdViewProvider adViewProvid
 
     private final AdvertsLoader.Callback advertsLoadedCallback = new AdvertsLoader.Callback() {
         @Override
-        public void onAdvertsLoaded(List<AdvertBreak> advertBreaks) {
+        public void onAdvertsLoaded(List<AdvertBreak> breaks) {
             loadingAds = null;
-            adPlaybackState = AdvertPlaybackState.from(advertBreaks);
+            advertBreaks.clear();
+            advertBreaks.addAll(breaks);
```

Comment:
If we make `advertBreaks` non final then we can have an empty list to start with (`Collections.emptyList()`) and here we can make a copy of the provided list and reassign it (`advertBreaks  = new ArrayList<>(list)`).
Which would make it more thread safe

Reply:
f76ddba